src/cclib/io/wfxwriter.py

```diff
@@ -0,0 +1,475 @@
+
+    def _mo_energies(self):
+        """List of MO Energies."""
+        mo_energies = []
+        alpha_elctrons = self._no_alpha_electrons()
+        beta_electrons = self._no_beta_electrons()
+        for mo_energy in self.ccdata.moenergies[0][:alpha_elctrons]:
+            mo_energies.append('{:22.11E}'.format(
+                utils.convertor(mo_energy, 'eV', 'hartree')))
+        if self.ccdata.mult > 1:
+            for mo_energy in self.ccdata.moenergies[1][:beta_electrons]:
+                mo_energies.append('{:22.11E}'.format(
+                    utils.convertor(mo_energy, 'eV', 'hartree')))
+        return mo_energies
+
+    def _mo_spin_types(self):
+        """MO Spin Types."""
+        spin_types = []
+        electrons = self._no_electrons()
+        alpha = self._no_alpha_electrons()
+        beta = self._no_beta_electrons()
+        if len(self.ccdata.homos) == 1:
+            spin_types += ['Alpha and Beta'] * int(electrons / 2) +\
+                            ['Alpha'] * (electrons % 2)
+        else:
+            spin_types += ['Alpha'] * alpha +\
+                            ['Beta'] * beta
+        return spin_types
+
+    pi3 = (2.0 / numpy.pi) ** 3
+
+    # Precomputed values for l+m+n/
+    _L = dict(
+        [(prim_type, 0) for prim_type in range(1, 2)] +   # s
+        [(prim_type, 1) for prim_type in range(2, 5)] +   # p
+        [(prim_type, 2) for prim_type in range(5, 11)] +  # d
+        [(prim_type, 3) for prim_type in range(11, 21)] + # f
+        [(prim_type, 4) for prim_type in range(21, 36)]   # g
+    )
+
+    # Precomputed values for ((2l-1)!! * (2m-1)!! * (2n-1)!!).
+    # Ref. Molden2AIM
+    _M = dict(
+        [(L, 1) for L in range(1, 5)] +
+        [(L, 9) for L in range(5, 8)] +
+        [(L, 1) for L in range(8, 11)] +
+        [(L, 225) for L in range(11, 14)] +
+        [(L, 9) for L in range(14, 20)] +
+        [(L, 1) for L in range(20, 21)] +
+        [(L, 11025) for L in range(21, 24)] +
+        [(L, 225) for L in range(24, 30)] +
+        [(L, 81) for L in range(30, 33)] +
+        [(L, 9) for L in range(33, 36)]
+    )
+
+    def _normalize(self, prim_type, alpha=1.0):
+        """Normalization factor for Cartesian Gaussian Functions.
+
+        N**4 = (2/pi)**3 * 2**(l+m+n) * alpha**(3 + 2(l+m+n)) /
+                            ((2l-1)!! * (2m-1)!! * (2n-1)!!)**2
+            = (2/pi)**3 * 2**(L) * alpha**(3 + 2L) /
+                            M**2,
+        L = l+m+n,
+        M = ((2l-1)!! * (2m-1)!! * (2n-1)!!)
+        """
+        L = self._L[prim_type]
+        M = self._M[prim_type]
+        norm_four = self.pi3 * 2**(4*L) * alpha**(3+2*L) / M
+        norm = numpy.power(norm_four, 1/4.0)
+        return norm
+
+    def _rearrange_mocoeffs(self, mocoeffs):
+        """Rearrange cartesian F functions in mocoeffs.
+        Expected order:
+        xxx, yyy, zzz, xyy, xxy, xxz, xzz, yzz, yyz, xyz
+        cclib's order:
+        XXX, YYY, ZZZ, XXY, XXZ, YYX, YYZ, ZZX, ZZY, XYZ
```

Comment:
This particular reordering is for `GAMESS`. I got it by doing a bit of reverse engineering on `wfn` files (which are the old style `wfx` files) generated using `GAMESS`.